diff --git a/crypto/riscv64cpuid.pl b/crypto/riscv64cpuid.pl
index 5dcdc5c584..7751ad50d9 100644
--- a/crypto/riscv64cpuid.pl
+++ b/crypto/riscv64cpuid.pl
@@ -94,8 +94,8 @@ $code .= <<___;
 .globl riscv_vlen_asm
 .type riscv_vlen_asm,\@function
 riscv_vlen_asm:
-    csrr $ret, vlenb
-    slli $ret, $ret, 3
+    csrr a0, 0xc22
+    slli a0, a0, 3
     ret
 .size riscv_vlen_asm,.-riscv_vlen_asm
 ___
